docs/latest/design/extending-envoy-gateway.md

```diff
@@ -120,46 +122,168 @@ spec:
+// an extension
+message HTTPListenerModifyResponse {
+    envoy.config.listener.v3.Listener listener = 1;
+}
+```
 
-Extending Envoy Gateway by using an external extension server which makes use of hook points in Envoy Gateway does come with some known issues that will have to be worked around.
+### Post xDS Translation Insert Hook
 
-- Extension server handling its own state such as in the scenario where an extension establishes watches its own CRDs in the cluster can lead to a scenario where
-ensuring that the extension and Envoy Gateway are in-sync becomes a challenge. There are two potential solutions to this problem.
-  - Option 1: Create a new External Provider mechanism where an extension can take over all provider responsibility for Envoy Gateway. The current functionality of the provider would need to become a Library that EG would make use of 
-to watch gateway API resources. If an extension wanted to take over the responsibility of watching everything, it would then become an "External Provider" and could use the same library Envoy Gateway would be using and add watches for 
-all new resources introduced by the extension. Extension developers would get the benefit of re-using the community provided Gateway-API common logic and would allow an extension to have control over when reconciliation occurs in Envoy Gateway.
-  - Option 2: Make an addition to Envoy Gateway's Gateway-API provider so that it listens to Custom Resource API Groups/Kinds that are introduced by extension and trigger a validation hook when reconciling these resources 
-so the extension can do early validation of its resources. Envoy Gateway would then pass these resources back to the extension whenever a hook point is called. This way the extension does not need to create any watches on resources in the 
-Kubernetes API Server. This solution has the added benefit of extension developers not needing to take control over resource watching from Envoy Gateway and is less operationally complex.
+The Post Translation Insert Hook allows an extension to return a list of clusters and secrets that should be injected into the xDS config. This allows for clusters that may change along with extension-specific configuration to be dynamically created rather than using custom bootstrap config which would be sufficient for clusters that are static and not prone to have their configurations changed. An example of how this may be used is to inject a cluster that will be used by an ext_authz HTTP filter created by the extension. This Hook is always executed when an extension is loaded. If the extension does not return any clusters/secrets then this is not considered an error and nothing is inserted.
+
+```go
+// This Hook does not take any inputs but expects a list of clusters and/or secrets to be injected. Returning empty lists of clusters
+// and/or secrets is equivalent to a no-op.
+message Empty {
```

Comment:
To support future, it might be better to just have a properly named message with an empty body.

```protobuf
message PostTranslationInsertRequest {}
```